src/pip/_internal/utils/misc.py

```diff
@@ -765,7 +765,7 @@ def warn_if_run_as_root() -> None:
         "conflicting behaviour with the system package manager, possibly "
         "rendering your system unusable. "
         "It is recommended to use a virtual environment instead: "
-        "https://pip.pypa.io/warnings/venv. "
+        "https://pip.pypa.io/warnings/venv ."
```

Comment:
TBH, the space looks ugly. In my opinion, it's the terminal's code for guessing what constitutes a URL that is wrong here (how often does a URL end in a dot?)

```suggestion
        "(see https://pip.pypa.io/warnings/venv)"
```

(plus, remove the colon from the line above, but I don't know how to get Github's "suggestion" mechanism to include that).

If the terminal still can't correctly find the URL, then IMO it's not our problem.

Reply:
As long as a URL _can_ end in a dot, the code for parsing the URL is valid and correct. The page that this is supposed to point does not end with a dot so it is currently wrong.

All fully-qualified domain names end with a dot. I could not find any restrictions on on a path ending with a dot? Sites like Wikipedia treat it as a valid URL.

The screenshot is from GitHub actions and not exactly from the terminal (I have edited the PR description to reflect the same).

Reply:
This seems the relevant spec: https://datatracker.ietf.org/doc/html/rfc3986/#appendix-C ?

It recommends using double quotes, e.g. `"https://pip.pypa.io/warnings/venv"`, or angled brackets, e.g. `"<https://pip.pypa.io/warnings/venv>"`. I don't have a strong opinion on if pip should adopt either of these conventions in displaying URLs in user facing messages, but if we do it should be consistent everywhere.

It also says software that accepts typed URIs should attempt to recognize these situations:

>   For robustness, software that accepts user-typed URI should attempt
>   to recognize and strip both delimiters and embedded whitespace.